File: STUST Java 2023-12-01/Student.java
public class Student {
    // Attributes
    private String name;
    private int age;
    private String studentId;
    private double gpa;

    // Constructor
    public Student(String name, int age, String studentId, double gpa) {
        this.name = name;
        this.age = age;
        this.studentId = studentId;
        this.gpa = gpa;
    }

    // Getter methods
    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public String getStudentId() {
        return studentId;
    }

    public double getGpa() {
        return gpa;
    }

    // Setter methods
    public void setName(String name) {
        this.name = name;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public void setStudentId(String studentId) {
        this.studentId = studentId;
    }

    public void setGpa(double gpa) {
        this.gpa = gpa;
    }

    // Display method
    public void displayStudentInfo() {
        System.out.println("Name: " + name);
        System.out.println("Age: " + age);
        System.out.println("Student ID: " + studentId);
        System.out.println("GPA: " + gpa);
    }

    // Example of usage
    public static void main(String[] args) {
        // Creating a new student object
        Student student1 = new Student("John Doe", 20, "S12345", 3.5);

        // Displaying student information
        System.out.println("Student Information:");
        student1.displayStudentInfo();

        // Modifying student information
        student1.setAge(21);
        student1.setGpa(3.8);

        // Displaying updated information
        System.out.println("\nUpdated Student Information:");
        student1.displayStudentInfo();
    }
}

File: STUST Python 2023-12-01/Student.py


class Student:
    # Constructor
    def __init__(self, name, age, student_id, gpa):
        self.name = name
        self.age = age
        self.student_id = student_id
        self.gpa = gpa


    @property
    def student_name(self):
        print(f'"{self.name}" was accessed.')
        return self.name

    @student_name.setter
    def student_name(self, value):
        print(f'"{self.name}" is now "{value}".')
        self.name = value

    @student_name.deleter
    def student_name(self): 
        print(f'"{self.name}" was deleted')
        del self.name

# Example of usage
if __name__ == "__main__":
    # Creating a new student object
    student1 = Student("John Doe", 20, "S12345", 3.5)

    print(student1.student_name)
    student1.student_name = "Peter Pan"
    del student1.student_name
    print(student1.student_name)

    # Displaying student information
    #print("Student Information:")
    #student1.display_student_info()

    # Displaying updated information
    #print("\nUpdated Student Information:")
    #student1.display_student_info()
